app/models/facility_group.rb

```diff
@@ -101,4 +105,25 @@ def state_region
                                                       region_type: Region.region_types[:state],
                                                       reparent_to: organization.region)
   end
+
+  def update_blocks
+    create_blocks if add_blocks.present?
+    destroy_blocks if remove_blocks.present?
+  end
+
+  def create_blocks
+    add_blocks.map do |block|
+      Region.create(
+        name: block,
+        region_type: Region.region_types[:block],
+        reparent_to: region
+      )
+    end
+  end
+
+  def destroy_blocks
+    remove_blocks.map do |id|
+      Region.destroy(id) if Region.find(id) && Region.find(id).children.empty?
+    end
+  end
```

Comment:
I'm concerned about using callbacks for this.  This seems like something better handled by the controller when blocks are added or removed.